src/server/routers/institution/instance/index.ts

```diff
@@ -115,266 +96,48 @@ export const instanceRouter = createTRPCRouter({
-      }),
-    )
-    .query(async ({ ctx, input: { params } }) => {
-      const allocationData = await getAllAllocationData(ctx.db, params);
-
-      const byStudent = allocationData
-        .map((allocation) => ({
-          student: {
-            id: allocation.student.user.id,
-            name: allocation.student.user.name!,
-            email: allocation.student.user.email!,
-            ranking: allocation.studentRanking,
-          },
-          project: {
-            id: allocation.project.id,
-            title: allocation.project.title,
-          },
-          supervisor: {
-            id: allocation.project.supervisor.user.id,
-            name: allocation.project.supervisor.user.name!,
-          },
-        }))
-        .sort((a, b) => a.student.id.localeCompare(b.student.id));
-
-      const byProject = allocationData
-        .map((allocation) => ({
-          project: {
-            id: allocation.project.id,
-            title: allocation.project.title,
-            capacityLowerBound: allocation.project.capacityLowerBound,
-            capacityUpperBound: allocation.project.capacityUpperBound,
-          },
-          supervisor: {
-            id: allocation.project.supervisor.user.id,
-            name: allocation.project.supervisor.user.name!,
-          },
-          student: {
-            id: allocation.student.user.id,
-            name: allocation.student.user.name,
-            ranking: allocation.studentRanking,
-          },
-        }))
-        .sort((a, b) => a.project.id.localeCompare(b.project.id));
-
-      // TODO: sort out duplicate rows (group or remove)
-      const bySupervisor = allocationData
-        .map((allocation) => ({
-          project: {
-            id: allocation.project.id,
-            title: allocation.project.title,
-          },
-          supervisor: {
-            id: allocation.project.supervisor.user.id,
-            name: allocation.project.supervisor.user.name!,
-            email: allocation.project.supervisor.user.email!,
-            allocationLowerBound:
-              allocation.project.supervisor.supervisorInstanceDetails[0]
-                .projectAllocationLowerBound,
-            allocationTarget:
-              allocation.project.supervisor.supervisorInstanceDetails[0]
-                .projectAllocationTarget,
-            allocationUpperBound:
-              allocation.project.supervisor.supervisorInstanceDetails[0]
-                .projectAllocationUpperBound,
-          },
-          student: {
-            id: allocation.student.user.id,
-            name: allocation.student.user.name,
-            ranking: allocation.studentRanking,
-          },
-        }))
-        .sort((a, b) => a.supervisor.id.localeCompare(b.supervisor.id));
-
-      return { byStudent, byProject, bySupervisor };
+    .query(async ({ input: { params } }) => {
+      return await getProjectAllocationsUseCase({ params });
     }),
 
```

Comment:
The extra input is quite annoying and ugly. With a bit of cleverness, you can get this down to 2 lines:

```suggestion
  projectAllocations: instanceAdminProcedure.query(async ({ ctx }) => {
    return await getProjectAllocationsUseCase({ params: ctx.instance.params });
  }),
```
Since we pull the params of the `ctx.instance` they have the right type. In general, you can do this for all your other admin procedures. I've got a version of the file where I implemented a lot of these and it really trims them down. Most of them are a very neat 2 line thing. I won't add suggestions for those for the sake of noise, but I can provide a patch file.